File: Assets/PythonMCP/modules/object_picker_module.py
import requests
import json
from typing import Dict, List

class ObjectPickerModule:
    def __init__(self, base_url: str):
        self.base_url = base_url
    
    def get_object_picker_options(self, object_path: str) -> Dict:
        """Получает 10 вариантов для Object Picker"""
        try:
            if not object_path:
                return {
                    "success": False,
                    "action": "object_picker_options",
                    "error": "object_path is required"
                }
            
            # Получаем иерархию для генерации вариантов
            hierarchy_response = requests.get(f"{self.base_url}/scene")
            hierarchy_response.raise_for_status()
            hierarchy = hierarchy_response.json()
            
            # Генерируем список всех объектов для Object Picker
            all_objects = self._extract_all_objects(hierarchy.get("rootObjects", []))
            
            # Фильтруем и берем первые 10 объектов
            picker_options = all_objects[:10]
            
            return {
                "success": True,
                "action": "object_picker_options",
                "data": {
                    "object_path": object_path,
                    "options": picker_options
                }
            }
            
        except requests.exceptions.RequestException as e:
            return {
                "success": False,
                "action": "object_picker_options",
                "error": f"Request error: {str(e)}"
            }
        except json.JSONDecodeError as e:
            return {
                "success": False,
                "action": "object_picker_options",
                "error": f"JSON decode error: {str(e)}"
            }
    
    def _extract_all_objects(self, root_objects: List, parent_path: str = "") -> List[Dict]:
        """Рекурсивно извлекает все объекты из иерархии"""
        objects = []
        
        for obj in root_objects:
            if isinstance(obj, dict):
                name = obj.get("name", "Unknown")
                path = f"{parent_path}/{name}" if parent_path else name
                
                # Добавляем текущий объект
                objects.append({
                    "name": name,
                    "path": path,
                    "type": "GameObject",
                    "components": obj.get("components", [])
                })
                
                # Рекурсивно обрабатываем дочерние объекты
                children = obj.get("children", [])
                if children:
                    objects.extend(self._extract_all_objects(children, path))
        
        return objects

File: Assets/Editor/SceneAPI/Modules/Objects/ObjectPickerModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using Newtonsoft.Json;
using UnityEngine;

namespace SceneAPI.Modules
{
    public static class ObjectPickerModule
    {
        public static string Execute(HttpListenerContext context)
        {
            try
            {
                string objectPath = context.Request.QueryString["path"] ?? "";
                string componentName = context.Request.QueryString["component"] ?? "";
                string parameterName = context.Request.QueryString["parameter"] ?? "";

                var result = new List<object>();

                // Получаем все объекты сцены
                GameObject[] allObjects = UnityEngine.Object.FindObjectsOfType<GameObject>();

                // Если указан путь к объекту, начинаем поиск от него
                GameObject startObject = null;
                if (!string.IsNullOrEmpty(objectPath))
                {
                    startObject = GameObjectUtilities.FindGameObjectByPath(objectPath);
                }

                // Фильтруем объекты по критериям
                var filteredObjects = allObjects.AsEnumerable();

                if (startObject != null)
                {
                    // Ищем дочерние объекты
                    filteredObjects = GetChildrenRecursive(startObject);
                }

                if (!string.IsNullOrEmpty(componentName))
                {
                    // Фильтруем по наличию компонента
                    filteredObjects = filteredObjects.Where(obj =>
                        obj.GetComponent(componentName) != null);
                }

                // Берем первые 10 объектов
                var selectedObjects = filteredObjects.Take(10);

                foreach (var obj in selectedObjects)
                {
                    var objInfo = new
                    {
                        name = obj.name,
                        path = GameObjectUtilities.GetGameObjectPath(obj),
                        type = "GameObject",
                        components = obj.GetComponents<Component>()
                            .Where(c => c != null)
                            .Select(c => c.GetType().Name)
                            .ToArray()
                    };

                    result.Add(objInfo);
                }

                return JsonConvert.SerializeObject(new
                {
                    success = true,
                    count = result.Count,
                    objects = result
                }, Formatting.Indented);
            }
            catch (Exception ex)
            {
                return JsonConvert.SerializeObject(new
                {
                    success = false,
                    error = $"Error getting object picker options: {ex.Message}"
                });
            }
        }

        private static IEnumerable<GameObject> GetChildrenRecursive(GameObject parent)
        {
            var children = new List<GameObject> { parent };

            for (int i = 0; i < parent.transform.childCount; i++)
            {
                var child = parent.transform.GetChild(i).gameObject;
                children.AddRange(GetChildrenRecursive(child));
            }

            return children;
        }
    }
}